src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -202,7 +202,22 @@ private function updateOrderDetails(Order $order, Cart $cart, int $computingPrec
             $cartProduct = $this->getProductFromCart($cartProducts, (int) $orderDetail->product_id, (int) $orderDetail->product_attribute_id);
 
             $unitPriceTaxExcl = (float) $cartProduct['price_with_reduction_without_tax'];
-            $unitPriceTaxIncl = (float) $cartProduct['price_without_reduction'];
+            $unitPriceTaxIncl = (float) $cartProduct['price_with_reduction'];
```

Comment:
when the product has reduction, wrong price was taken
The orderDetail was incorrect after that

Reply:
Can you add some checks in behat tests then please? Because there are other pending PRs that modify this function so we must avoid any future regression